userbenchmark/ddp_experiments/__init__.py

```diff
@@ -489,6 +469,222 @@ def get_backend_name(model_args):
+
+        backend_name = get_backend_name(model_args)
+        config = {
+            "nodes": nodes,
+            "model_name": model_name,
+            "backend": backend_name,
+        }
+        return ExperimentParams(config, args_copy, copied_model_args, is_reference=fsdp_is_reference(backend_name))
+
+    def get_hf_GPT2_large_config(nodes, model_args):
+        model_path = "torchbenchmark.models.hf_GPT2_large.Model"
+        args_copy, copied_model_args = generic_setup(nodes, model_args)
+        copied_model_args.extend(["--distributed_wrap_fn", "userbenchmark.ddp_experiments.apply_fsdp_hf_GPT2_large"])
+
+        # assuming 8 gpus per node
+        # 8 sometimes passes / sometimes fails depending on config
+        batch_size_per_nodes = {1: 6, 2: 6, 4: 6, 8: 6}
+
+        assert nodes in batch_size_per_nodes
+        args_copy.batch_size = batch_size_per_nodes[nodes]
+        args_copy.model = model_path
+
+        backend_name = get_backend_name(model_args)
+        config = {
+            "nodes": nodes,
+            "model_name": model_name,
+            "backend": backend_name,
+        }
+        return ExperimentParams(config, args_copy, copied_model_args, is_reference=fsdp_is_reference(backend_name))
+
+    def get_timm_VIT_large_config(nodes, model_args):
+        model_path = "torchbenchmark.models.timm_vision_transformer_large.Model"
+        args_copy, copied_model_args = generic_setup(nodes, model_args)
+        copied_model_args.extend(["--distributed_wrap_fn", "userbenchmark.ddp_experiments.apply_fsdp_timm_VIT_large"])
+
+        # assuming 8 gpus per node
+        # 4, 8: tried bs = 32, and this OOMed.
+        batch_size_per_nodes = {1: 24, 2: 24, 4: 24, 8: 24}
+
+        assert nodes in batch_size_per_nodes
+        args_copy.batch_size = batch_size_per_nodes[nodes]
+        args_copy.model = model_path
+
+        backend_name = get_backend_name(model_args)
+        config = {
+            "nodes": nodes,
+            "model_name": model_name,
+            "backend": backend_name,
+        }
+        return ExperimentParams(config, args_copy, copied_model_args, is_reference=fsdp_is_reference(backend_name))
+
+    model_configs = {
+        "timm_vision_transformer_large": get_timm_VIT_large_config,
+        "hf_GPT2_large": get_hf_GPT2_large_config,
+        "hf_T5_large": get_hf_T5_large_config,
+    }
+
+    model_args_configs = [
+        [],  # no args = pure eager baseline
+        ["--torchdynamo", "inductor"],
+    ]
+    # run the 8-node version first so that all the caches get warmed up at the same time.
+    node_list = [8, 4, 2, 1]
+
+    experiments = []
+    for i in range(args.repeat):
+        for nodes in node_list:
+            for model_name, config_generator in model_configs.items():
+                for model_args in model_args_configs:
+                    experiments.append(config_generator(nodes, model_args))
+
+    allocation_nodes = max(node_list)
+    executor.update_parameters(
```

Comment:
does this need to be duplicated from below?

Reply:
de-duped a bunch of this